lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2158,17 +2164,17 @@ public function createEntity($className, array $data, &$hints = [])
 
             $this->identityMap[$class->getRootClassName()][$idHash] = $entity;
 
-            if ($entity instanceof NotifyPropertyChanged) {
-                $entity->addPropertyChangedListener($this);
-            }
-
             $overrideLocalValues = true;
         }
 
         if ( ! $overrideLocalValues) {
             return $entity;
         }
 
+        if ($entity instanceof NotifyPropertyChanged) {
```

Comment:
This was moved, because the `addPropertyChangedListener()` call must happen both for initialized and un-initialized proxies once the `UnitOfWork` starts touching them